pyproject.toml

```diff
@@ -48,11 +49,11 @@ line_length = 120
 
 [tool.black]
 line-length = 120
-target-version = ["py311"]
+target-version = ["py37"]
```

Comment:
Why 3.7 here and not 3.12?

Reply:
I meant that because below is defined in `pyproject.toml`:

```toml
classifiers = [
    "License :: OSI Approved :: Apache Software License",
    "Programming Language :: Python",
    "Programming Language :: Python :: 3",
    "Programming Language :: Python :: 3.7",
    "Programming Language :: Python :: 3.8",
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
    "Programming Language :: Python :: 3.12",
]
```

However, I checked black official documentation again, below was correct setting:

```toml
target-version = ["py37", "py38", "py39", "py310", "py311", "py312"]
```

cf. [The basics - Black 24.4.1 documentation](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#command-line-options)


And also in case of Ruff, it seems we should define as below:

```diff
+ [project]
+ requires-python = ">=3.7"

[tool.ruff]
line-length = 120
- target-version = "py37"
```

cf. [Settings - Ruff](https://docs.astral.sh/ruff/settings/#target-version)

Reply:
@bmorrissirromb I added one more revision to fix so. If you won't so, please fix by yourself.

And, additionally:

```diff
+ [project]
+ name = "rdklib"
+ requires-python = ">=3.7"

[tool.ruff]
line-length = 120
- target-version = "py37"
```

The property: `name` seems to be required in `project`: [schemastore/src/schemas/json/pyproject.json at master · SchemaStore/schemastore](https://github.com/SchemaStore/schemastore/blob/15c0a5141e10d0eb77f53db6d2222a5c35839575/src/schemas/json/pyproject.json#L148)